ably/options.go

```diff
@@ -59,13 +64,24 @@ var defaultOptions = clientOptions{
 }
 
 func defaultFallbackHosts() []string {
-	return []string{
-		"a.ably-realtime.com",
-		"b.ably-realtime.com",
-		"c.ably-realtime.com",
-		"d.ably-realtime.com",
-		"e.ably-realtime.com",
+	return endpointFallbacks("main", "ably-realtime.com")
+}
+
+func getEndpointFallbackHosts(endpoint string) []string {
+	if match := regexp.MustCompile(`^nonprod:(.*)$`).FindStringSubmatch(endpoint); match != nil {
+		namespace := match[1]
+		return endpointFallbacks(namespace, "ably-realtime-nonprod.com")
 	}
```

Comment:
compiling a regexp in Go is very expensive. (But once a regexp is compiled, searching is fast and guaranteed to execute in a linear time).
So we usually compile the regexp at init time.

```suggestion

var nonProdRegexp = regexp.MustCompile(`^nonprod:(.*)$`)

func getEndpointFallbackHosts(endpoint string) []string {
	if match := nonProdRegexp.FindStringSubmatch(endpoint); match != nil {
		namespace := match[1]
		return endpointFallbacks(namespace, "ably-realtime-nonprod.com")
	}
```

This also has the benefit that if we have a typo in the regexp, so that it fails to compile, this will always panic when the program starts (which is what we want), rather than only when getEndpointFallbackHosts is called.

Reply:
Fixed up in ee7c1b1bd8016bb2ec1029aeebc39af977cd1f5a